### PR TITLE
Entitlement api

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Describe:
This PR adds the Entitlement Management API implementation for the following APIs.

Screenshot 2024-12-06 at 11 14 37

Screenshot 2024-12-06 at 11 14 30

Postman collection to test the APIs:
- [Entitlements.postman_collection.json](https://github.com/user-attachments/files/18033238/Entitlements.postman_collection.json)

### Related Issues:
- https://github.com/wso2/product-is/issues/21959
- https://github.com/wso2/product-is/issues/21675

### CODE REVIEW

File: components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.ws.rs</groupId>
+            <artifactId>javax.ws.rs-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (Thumimku) [inline components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.common/pom.xml:59]: EOF check other places also

File: components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.v1/src/gen/java/org/wso2/carbon/identity/api/server/entitlement/v1/factories/EntitlementsApiServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import org.wso2.carbon.identity.api.server.entitlement.v1.EntitlementsApiService;
+import org.wso2.carbon.identity.api.server.entitlement.v1.impl.EntitlementsApiServiceImpl;
+
+public class EntitlementsApiServiceFactory {
+
+   private final static EntitlementsApiService service = new EntitlementsApiServiceImpl();
+
+   public static EntitlementsApiService getEntitlementsApi()
+   {
```
Comment (lashinijay) [inline components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.v1/src/gen/java/org/wso2/carbon/identity/api/server/entitlement/v1/factories/EntitlementsApiServiceFactory.java:29]:
```suggestion
   public static EntitlementsApiService getEntitlementsApi() {
```

File: components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.v1/src/main/resources/META-INF/cxf/entitlement-cxf.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (lashinijay) [inline components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.v1/src/main/resources/META-INF/cxf/entitlement-cxf.xml:20]: Shall we remove this file as there is no usage of this anymore.

File: components/entitlement/org.wso2.carbon.identity.entitlement.mgt.endpoint/src/main/webapp/WEB-INF/web.xml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider
+            </param-value>
+        </init-param>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>CXFServlet</servlet-name>
+        <url-pattern>/*</url-pattern>
+    </servlet-mapping>
+</web-app>
```
Comment (lashinijay) [inline components/entitlement/org.wso2.carbon.identity.entitlement.mgt.endpoint/src/main/webapp/WEB-INF/web.xml:56]: A new line is missing

File: components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.common/pom.xml (hunk 1 above)
Comment (mpmadhavig) [reply components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.common/pom.xml]: beaad0bcbef51e595091ef9e1217c2a4077fbc03

File: components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.v1/src/main/resources/META-INF/cxf/entitlement-cxf.xml (hunk 3 above)
Comment (mpmadhavig) [reply components/entitlement/org.wso2.carbon.identity.api.server.entitlement/org.wso2.carbon.identity.api.server.entitlement.v1/src/main/resources/META-INF/cxf/entitlement-cxf.xml]: beaad0bcbef51e595091ef9e1217c2a4077fbc03

File: components/entitlement/org.wso2.carbon.identity.entitlement.mgt.endpoint/src/main/webapp/WEB-INF/web.xml (hunk 4 above)
Comment (mpmadhavig) [reply components/entitlement/org.wso2.carbon.identity.entitlement.mgt.endpoint/src/main/webapp/WEB-INF/web.xml]: beaad0bcbef51e595091ef9e1217c2a4077fbc03

**Review by Thumimku — APPROVED** [review]

**Review by Thumimku — APPROVED** [review]

**Pull request merged**